internal/api/v1beta1/groupversion_info.go

```diff
@@ -33,4 +35,15 @@ var (
 
 	// AddToScheme adds the types in this group-version to the given scheme.
 	AddToScheme = SchemeBuilder.AddToScheme
+
+	// Group is the CRD group
+	Group = "theketch.io"
 )
+
+func init() {
```

Comment:
I'm a bit concerned with such an initialization:
1. both ketch-cli and ketch-manager import this package, so the CUSTOM_GROUP variable works for both. Let's avoid this behavior. Ketch-cli should always work with the `theketch.io` group.
2. Shipa imports this package too and the only way to set `shipa.io` group is to set the CUSTOM_GROUP env, this looks tricky to me
3. A lot of initialization is done in [main.go](https://github.com/shipa-corp/ketch/blob/main/cmd/manager/main.go#L49), and usually it is a good point to understand what the parameters to configure ketch are. For example, one can open main.go and see that webhooks can be disabled with `--disable-webhooks`. The suggestion is to use CUSTOM_GROUP in main.go

so can we go with something like this:
// groupversion_info.go
```golang
const (
	TheKetchGroup = "theketch.io"
)

type SchemeOptions struct {
	group string
}

type SchemeOption func(opts *SchemeOptions)

func WithGroup(group string) SchemeOption {
	return func(opts *SchemeOptions) {
		opts.group = group
	}
}

func defaultSchemeOptions() SchemeOptions {
	return SchemeOptions{group: TheKetchGroup}
}

// AddToScheme can be easily consumed by ketch-cli and by default it uses `theketch.io`
// and it can be consumed by shipa: AddToScheme(WithGroup("shipa.io")) 

func AddToScheme(opts ...SchemeOption) func(s *runtime.Scheme) error {
	options := defaultSchemeOptions()
	for _, o := range opts {
		o(&options)
	}
	groupVersion := schema.GroupVersion{Group: options.group, Version: "v1beta1"}
	builder := &scheme.Builder{GroupVersion: groupVersion}
	return builder.AddToScheme
}

```

// main.go
```golang
func main() {
	_ = clientgoscheme.AddToScheme(scheme)

	group := ketchv1.TheKetchGroup
	if len(os.Getenv("CUSTOM_GROUP")) > 0 {
		group = os.Getenv("CUSTOM_GROUP")
	}
	_ = ketchv1.AddToScheme(ketchv1.WithGroup(group))(scheme)

		// +kubebuilder:scaffold:scheme
	var metricsAddr string
	var enableLeaderElection bool
	var disableWebhooks bool
	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true,
		"Enable leader election for controller manager. "+
			"Enabling this will ensure there is only one active controller manager.")
	flag.BoolVar(&disableWebhooks, "disable-webhooks", false, "Disable webhooks.")
	flag.Parse()
```

I think we should avoid complex `init` functions(this one takes arguments from env variable and has side-effects) and global variables (`Group`) in general

Reply:
@aleksej-paschenko Thanks for the help. I moved the group logic out of `init()` and into `main()` as you suggested and switched from an environment variable to a `flag`. I'm curious as to how we'll end up implementing this - does a flag work?

Reply:
it should work, we can add the flags there https://github.com/shipa-corp/shipa/pull/927/files#diff-f305fab607f0d165929a8f357219b61fe04fcae48fc9cdcb09ef1e86fad3c3c0R25